crates/types/src/lib.rs

```diff
@@ -522,6 +522,13 @@ impl WasmHeapType {
         }
     }
 
+    pub fn is_cont_type(&self) -> bool {
+        match self {
+            WasmHeapType::Cont | WasmHeapType::ConcreteCont(_) | WasmHeapType::NoCont => true,
+            _ => false,
+        }
```

Comment:
Use the `top` function here instead.

```suggestion
        match self.top() {
            WasmHeapTopType::Cont => true,
            _ => false,
        }
```